pyanaconda/modules/payloads/source/url/url.py

```diff
@@ -0,0 +1,145 @@
+#
+# Kickstart module for URL payload source.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core.constants import BASE_REPO_NAME
+from pyanaconda.core.signal import Signal
+from pyanaconda.core.util import ProxyString, ProxyStringError
+from pyanaconda.modules.common.errors import InvalidValueError
+from pyanaconda.modules.common.structures.payload import RepoConfigurationData
+from pyanaconda.modules.payloads.constants import SourceType, URLType
+from pyanaconda.modules.payloads.source.source_base import PayloadSourceBase
+from pyanaconda.modules.payloads.source.url.url_interface import URLSourceInterface
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class URLSourceModule(PayloadSourceBase):
+    """The URL source payload module."""
+
+    REPO_NAME_ID = 0
+
+    def __init__(self):
+        super().__init__()
+        self._repo_configuration = RepoConfigurationData()
+        self.repo_configuration_changed = Signal()
+
+        self._url_source_id = self._generate_source_id()
+        self._set_repo_config_name()
+
+        self._install = False
+        self.install_changed = Signal()
+
+    @classmethod
+    def _generate_source_id(cls):
+        source_id = cls.REPO_NAME_ID
+        cls.REPO_NAME_ID = cls.REPO_NAME_ID + 1
```

Comment:
This will generate same names for any subclasses of `URLSourceModule` if we ever create such classes.

Reply:
I'm not sure I understand you correctly. This is a class variable so it is shared between all subclasses with the same value. So if any subclass will generate new id it will increment the number for all the subclasses and parent class too?

Or are you talking about something else?

Reply:
Not exactly. If you try this script:

```
class X(object):
    id = 0

    @classmethod
    def get_id(cls):
        my_id = cls.id
        cls.id = cls.id + 1
        return my_id

class Y(X):
    pass

class Z(X):
    pass

for cls in [X, Y, Z, X, Y, Z, X, Y, Z]:
    print(cls.get_id())

```

The output is:
```
0
1
1
1
2
2
2
3
3
```

It should be a static method that will set up specifically `URLSourceModule.REPO_NAME_ID`. Also, please rename `REPO_NAME_ID` to `_repo_counter`. It is not a public constant.

Reply:
Interesting, I've tried this and it worked. I wonder what I did wrong before...
Fixed.